pulpcore/cli/rpm/content.py

```diff
@@ -285,10 +297,81 @@ def upload(
     chunk_size: int,
     **kwargs: t.Any,
 ) -> None:
-    """Create a content unit by uploading a file"""
-    body: t.Dict[str, t.Any]
+    """Create a content unit by uploading a file or files"""
     if isinstance(entity_ctx, PulpRpmPackageContext):
-        result = entity_ctx.upload(file=file, chunk_size=chunk_size, **kwargs)
+        directory = kwargs.get("directory")
+        if (not (file or directory)) or (file and directory):
+            raise click.ClickException(
+                _("You must specify one (and only one) of --file or --directory.")
+            )
+        if file:
+            result = entity_ctx.upload(file=file, chunk_size=chunk_size, **kwargs)
+        else:
+            # Create a temp-repo
+            dest_repo_ctx = kwargs["repository"]
+            tmp_repo_ctx = PulpRpmRepositoryContext(pulp_ctx)
+            body: t.Dict[str, t.Any] = {
+                "name": f"uploadtmp_{dest_repo_ctx.entity['name']}_{uuid4()}",
+                "retain_repo_versions": 1,
+                "autopublish": False,
+            }
+            result = tmp_repo_ctx.create(body=body)
```

Comment:
Obviously, we could provide a `--tmp-repository` parameter to circumvent the need for a creation permission. But that does not sound like a good idea. And running multiple of these operations with the same tmp storage won't give the desired results.
If we had a stable naming scheme, (e.g. `^tmp[/a]{8}$` or maybe `^tmp/.*$`) we could add a regex based access policy helper to allow anyone to create those repositories.